### DISCUSSION TITLE
Acute multianimal tattooed tail sessions

### DISCUSSION THREAD

**Discussion by jkbhagatio** [SainsburyWellcomeCentre/aeon_experiments#260] [General]
Following up from the progress in #141, we are starting to run three-animal sessions in B2-210 (AEON3) with tattooed tails for which we will test the inference of a SLEAP trained network. We will run these sessions under the 'presocial0.1' experiment. 

The plan is to run 30-60 minute sessions twice a week (Tuesdays and Fridays) with all three animals for the next 4 weeks, and assess the stability of the SLEAP trained network's inference over time.

The animals used for these sessions are:
- BAA-1103351 (dual band tattoo)
- BAA-1103352 (no tattoo)
- BAA-1103353 (single band tattoo)

The first sessions occurred on 2023-06-16.

BAA-1103353 had to be culled due to tail complications.

**Comment by lochhh** [comment]
The three-animal sessions are restarted with a new single-band-tattoo mouse and run under the 'multianimal-test' experiment in B2-210 (AEON3).

The animals used for these sessions are:
- BAA-1103351 (dual band tattoo)
- BAA-1103352 (no tattoo)
- CAA-1120139 (single band tattoo)

The first sessions occurred on 2023-07-04.

Again, CAA-1120139 had to be culled due to tail complications.

**Comment by lochhh** [comment]
The three-animal sessions are restarted the third time with a new single-band-tattoo mouse and run under the 'multianimal-test' experiment in B2-210 (AEON3).

The animals used for these sessions are:
- BAA-1103351 (dual band tattoo)
- BAA-1103352 (no tattoo)
- BAA-1103369 (single band tattoo)

The sessions occurred on 
- 2023-07-28
- 2023-08-01 
- 2023-08-03
- 2023-08-08
- 2023-08-11

**Reply by lochhh** [reply]
Training dataset consists of:
- 1120 single-animal frames from 2023-07-04 for BAA-1103351 (dual band tattoo) 
- 1008 single-animal frames from 2023-07-04 for BAA-1103352 (no tattoo) 
- 1126 single-animal frames from 2023-07-28 for BAA-1103369 (single band tattoo)
- 287 multi-animal frames from 2023-07-28

Resulting in models:
- AEON3B_pattern_tattoo3_1_topdown_top.centroid
- AEON3B_pattern_tattoo3_1_topdown_top.centered_instance_multiclass

Performance on 100 consecutive and randomly selected ground truth multi-animal frames from different sessions

| Stats/Dataset                | 2023-07-28      | 2023-08-01      | 2023-08-03      | 2023-08-08      | 2023-08-11      |  
|------------------------------|------------|------------|------------|------------|------------|
| Tracks identified            | 261        | 298        | 275        | 260        | 242        |
| Tracks correctly identified  | 248        | 297        | 274        | 252        | 223        |
| Total tracks                 | 300        | 300        | 300        | 300        | 300        |
| ID accuracy                  | 0.9502     | 0.9966     | 0.9964     | 0.9692     | 0.9215     |
| Frames with < 3 tracks       | 0.39       | 0.02       | 0.24       | 0.40       | 0.57       | 
| Frames with 2 missing tracks | 0.00       | 0.00       | 0.01       | 0.00       | 0.01       | 
| Frames with 1 missing track  | 0.39       | 0.02       | 0.23       | 0.40       | 0.56       | 
| Track occupancy              | [0.87 1. 0.74] | [1.   0.98 1.  ] | [0.79 1.   0.96] | [1.   0.82 0.78] | [0.96 0.65 0.81] | 

Performance on 100 non-consecutive and randomly selected ground truth multi-animal frames from different sessions

| Stats/Dataset                | 2023-07-28      | 2023-08-01      | 2023-08-03      | 2023-08-08      | 2023-08-11      |  
|------------------------------|------------|------------|------------|------------|------------|
| Tracks identified            | 233        | 243        | 259        | 256        | 254        |
| Tracks correctly identified  | 217        | 231        | 248        | 224        | 227        |
| Total tracks                 | 300        | 300        | 300        | 300        | 300        |
| ID accuracy                  | 0.9313     | 0.9506     | 0.9575     | 0.8750     | 0.8937     |
| Frames with < 3 tracks       | 0.59       | 0.49       | 0.38       | 0.42       | 0.43       | 
| Frames with 2 missing tracks | 0.08       | 0.08       | 0.03       | 0.02       | 0.03       | 
| Frames with 1 missing track  | 0.51       | 0.41       | 0.35       | 0.40       | 0.40       | 
| Track occupancy              | [0.82 0.84 0.67] | [0.84 0.75 0.84] | [0.94 0.83 0.82] | [0.92 0.83 0.81] | [0.91 0.83 0.8 ] |

**Comment by lochhh** [comment]
```mermaid
  flowchart TD
    A(AEON Data) --> B[/single_animal_frames/]
    A --> C[/multi_animal_frames/]
    B --> |curate| E[Auto labelling]
    C --> |curate| F[Manual labelling]
    F --> G[/training_dataset.slp/]
    E --> G
    G --> H[Train model]
    H --> I[/model.pb/]
    I --> J[Predict]
    J --> D[/predictions.slp/]
    C --> |all frames| J
    D --> K[Extract and correct random n low-scoring frames]
    D --> M[Extract and correct random n consecutive frames]
    K --> N[/low_GT.slp/]
    M --> O[/rand_GT.slp/]
    O --> P[Evaluate model]
    P --> Q{Is model good enough?}
    Q --> |yes| R(End training)
    Q --> |no| K
    N --> |merge| G
 ```

**Comment by lochhh** [comment]
```mermaid
flowchart TD
    raw_data(AEON Data) --> sa_frames[/single_animal_frames/]
    raw_data --> ma_frames[/multi_animal_frames/]
    sa_frames --> |all frames| proc_auto_label[Auto labelling]
    ma_frames --> |curate| proc_manual_label[Manual labelling]
    proc_auto_label --> |curate| training_dataset[/training_dataset.slp/]
    proc_manual_label --> training_dataset
    proc_auto_label --> |all frames| sa_gt[/single_animal_groundtruth.slp/]
    sa_gt --> proc_predict
    training_dataset --> proc_train_model[Train model]
    proc_train_model --> exported_model[/model.pb/]
    exported_model --> proc_predict[Predict]
    proc_predict --> ma_pr[/multi_animal_predictions.slp/]
    proc_predict --> sa_pr[/single_animal_predictions.slp/]
    ma_frames --> |all frames| proc_predict
    sa_pr --> proc_eval[Evaluate]
    ma_pr --> proc_eval
    proc_eval --> retrain{Is model good enough?}
    retrain --> |yes| exit_loop(End training)
    retrain --> |no| proc_correct[Extract and correct random n low-scoring frames]
    proc_correct --> corrected_gt[/low_score_groundtruth.slp/]
    corrected_gt --> |merge| training_dataset
```
